### PR TITLE
Requests mock

### PR DESCRIPTION
**Pull request by reginafcompton**
This PR handles issue #28

### CODE REVIEW

File: intermediate-python-testing.md [hunk 1]
```diff
@@ -361,3 +362,74 @@ Then, you need to patch the class you'd like to mock, and set its `return_value`
 mock_handle = mock.patch('path.to.CLASS_YOU_ARE_MOCKING')
 mock_handle.return_value = mocked_class
 ```
+
+## requests-mock
+
+Sometimes, your app sends requests to external URIs. And sometimes, these requests either have very little to do with the needs of your tests or fully interfere with them. [`requests-mock`](https://requests-mock.readthedocs.io/en/latest/overview.html) leaps into action!
```
Comment (hancush) [inline intermediate-python-testing.md]: great points. i'd add that it violates test isolation to rely on requests to an external service!

File: intermediate-python-testing.md [hunk 2]
```diff
@@ -361,3 +362,74 @@ Then, you need to patch the class you'd like to mock, and set its `return_value`
...
+
+`requests-mock` sets up [a transport adapter](http://docs.python-requests.org/en/latest/user/advanced/#transport-adapters) (a mechanism which defines how to handle HTTP requests). With a `requests-mock` adapter, you can load data into responses from specified URIs. How to use it? `requests-mock` comes with [multiple patterns for instantiating the `requests_mock.Mocker` class](https://requests-mock.readthedocs.io/en/latest/mocker.html#activation). For instance, use it as a context manager:
```
Comment (hancush) [inline intermediate-python-testing.md]:
```suggestion
`requests-mock` sets up [a transport adapter](http://docs.python-requests.org/en/latest/user/advanced/#transport-adapters) (a mechanism which defines how to handle HTTP requests). With a `requests-mock` adapter, requests to the specified URIs are automatically patched, giving you control of the data they return. How to use it? `requests-mock` comes with [multiple patterns for instantiating the `requests_mock.Mocker` class](https://requests-mock.readthedocs.io/en/latest/mocker.html#activation). For instance, use it as a context manager:
```

File: intermediate-python-testing.md [hunk 3]
```diff
@@ -361,3 +362,74 @@ Then, you need to patch the class you'd like to mock, and set its `return_value`
...
+
+```
+import requests
+import requests_mock
+
+def test_get_resp():
+    with requests_mock.Mocker() as m:
+        m.get("http://propane.org", text="that boy ain't right")
```
Comment (hancush) [inline intermediate-python-testing.md]: lol regina

File: intermediate-python-testing.md [hunk 4]
```diff
@@ -361,3 +362,74 @@ Then, you need to patch the class you'd like to mock, and set its `return_value`
...
+        rsp = requests.get("http://propane.org").text
+
+        assert rsp.text == "that boy ain't right"
+``` 
+
+Your app might hit variations of the same URI, however. `requests-mock` saves you the tedium of mocking each path. You can define an adaptor that matches multiple paths, [e.g., with Regex](https://requests-mock.readthedocs.io/en/latest/matching.html#regular-expressions). Let's return to the example above, but build out some of its functionality.
```
Comment (hancush) [inline intermediate-python-testing.md]:
```suggestion
Your app might hit variations of the same URI, such as different endpoints of an API. `requests-mock` saves you the tedium of mocking each path. You can define an adaptor that matches multiple paths, [e.g., with Regex](https://requests-mock.readthedocs.io/en/latest/matching.html#regular-expressions). Let's return to the example above, but build out some of its functionality.
```

File: intermediate-python-testing.md [hunk 5]
```diff
@@ -361,3 +362,74 @@ Then, you need to patch the class you'd like to mock, and set its `return_value`
...
 mock_handle.return_value = mocked_class
 ```
+
+## requests-mock
+
+Sometimes, your app sends requests to external URIs. And sometimes, these requests either have very little to do with the needs of your tests or fully interfere with them. [`requests-mock`](https://requests-mock.readthedocs.io/en/latest/overview.html) leaps into action!
+
+`requests-mock` sets up [a transport adapter](http://docs.python-requests.org/en/latest/user/advanced/#transport-adapters) (a mechanism which defines how to handle HTTP requests). With a `requests-mock` adapter, you can load data into responses from specified URIs. How to use it? `requests-mock` comes with [multiple patterns for instantiating the `requests_mock.Mocker` class](https://requests-mock.readthedocs.io/en/latest/mocker.html#activation). For instance, use it as a context manager:
+
+```
```
Comment (hancush) [inline intermediate-python-testing.md]:
```suggestion
```python
```

File: intermediate-python-testing.md [hunk 6]
```diff
@@ -361,3 +362,74 @@ Then, you need to patch the class you'd like to mock, and set its `return_value`
...
+``` 
+
+Your app might hit variations of the same URI, however. `requests-mock` saves you the tedium of mocking each path. You can define an adaptor that matches multiple paths, [e.g., with Regex](https://requests-mock.readthedocs.io/en/latest/matching.html#regular-expressions). Let's return to the example above, but build out some of its functionality.
+
+**`utils.py`**
+```python
+def get_result():
+    # query the database and return result
+
+def scrape_an_endpoint(endpoint):
```
Comment (hancush) [inline intermediate-python-testing.md]: i'm not sure i follow the arg being `endpoint`. based on the code in `tasks.py`, it looks like it should be `result_id` or something similar?

Comment (reginafcompton) [reply intermediate-python-testing.md]: yup, that's much more clear – variable names, important in the real world, important in the abstract

**Comment by reginafcompton** [comment]
@hancush - edits made! thanks for those nifty suggestions.

**Pull request merged**